File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        length = int(len(s)/2)
        
        a = s[0:length]
        # print(a)
        b = s[length:]
        # print(b)
        
        count_a = 0
        for i in range(0, len(a)):
            if a[i] in vowels:
                count_a += 1
                
        # print(count_a)
        
        count_b = 0
        for i in range(0, len(b)):
            if b[i] in vowels:
                count_b += 1
                
        # print(count_b)
        
        
        if count_a == count_b :
            return True
        else:
            return False

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        int cta = 0;
        for(int i = 0; i <s.length()/2; i++){
             if(s.charAt(i) == 'a'|| s.charAt(i) =='e'|| s.charAt(i) == 'i'|| s.charAt(i) =='o'|| s.charAt(i)== 'u' || s.charAt(i) =='A' || s.charAt(i) == 'E' || s.charAt(i) =='I' || s.charAt(i) == 'O' || s.charAt(i) =='U')
             {
                cta++;
            }
        }
        
        for(int i = s.length()/2; i <s.length(); i++){
            if(s.charAt(i) == 'a' || s.charAt(i) =='e' || s.charAt(i) == 'i' || s.charAt(i) =='o' || s.charAt(i)== 'u' || s.charAt(i) =='A' || s.charAt(i) == 'E' || s.charAt(i) =='I' || s.charAt(i) == 'O' || s.charAt(i) =='U')
            {
                cta--;
            }
        }
        
        return cta==0;
    }
}